src/datatypes/epgsql_codec_json.erl

```diff
@@ -17,18 +17,47 @@
 
 -define(JSONB_VERSION_1, 1).
 
-%% TODO: JSON encode/decode `fun Mod:Name/1` / `{Mod, Name}` as option.
-%% Shall not pass `fun(_) -> .. end`, because of hot code upgrade problems.
-init(_, _) -> [].
+%% JsonMod shall either be a module with encode/1 and decode/1, where:
+%%
+%% - encode/1 expects erlang-formatted JSON, e.g. a map or a proplist,
+%%   and returns a valid JSON binary string.
+%% - decode/1 expects a valid JSON binary string, and returns erlang-formatted JSON.
+%%
+%% or a tuple {Mod, {encode, Opts}, {decode, Opts}} where Mod implements encode/2 and
```

Comment:
do you think it makes sense to add extra tags `encode` and `decode`? Won't just `{Mod, EncodeOpts, DecodeOpts}` be enough?

Reply:
That'll definitely be better :+1: